wavesrv/pkg/sstore/updatebus.go

```diff
@@ -42,67 +41,70 @@ func (*PtyDataUpdate) UpdateType() string {
-	Info                     *InfoMsgType                       `json:"info,omitempty"`
-	ClearInfo                bool                               `json:"clearinfo,omitempty"`
-	Remotes                  []RemoteRuntimeState               `json:"remotes,omitempty"`
-	History                  *HistoryInfoType                   `json:"history,omitempty"`
-	Interactive              bool                               `json:"interactive"`
-	Connect                  bool                               `json:"connect,omitempty"`
-	MainView                 string                             `json:"mainview,omitempty"`
-	Bookmarks                []*BookmarkType                    `json:"bookmarks,omitempty"`
-	SelectedBookmark         string                             `json:"selectedbookmark,omitempty"`
-	HistoryViewData          *HistoryViewData                   `json:"historyviewdata,omitempty"`
-	ClientData               *ClientData                        `json:"clientdata,omitempty"`
-	RemoteView               *RemoteViewType                    `json:"remoteview,omitempty"`
-	ScreenTombstones         []*ScreenTombstoneType             `json:"screentombstones,omitempty"`
-	SessionTombstones        []*SessionTombstoneType            `json:"sessiontombstones,omitempty"`
-	OpenAICmdInfoChat        []*packet.OpenAICmdInfoChatMessage `json:"openaicmdinfochat,omitempty"`
-	AlertMessage             *AlertMessageType                  `json:"alertmessage,omitempty"`
-	ScreenStatusIndicators   []*ScreenStatusIndicatorType       `json:"screenstatusindicators,omitempty"`
-	ScreenNumRunningCommands []*ScreenNumRunningCommandsType    `json:"screennumrunningcommands,omitempty"`
-	UserInputRequest         *UserInputRequestType              `json:"userinputrequest,omitempty"`
-}
+type ModelUpdate []*ModelUpdateItem
 
 func (*ModelUpdate) UpdateType() string {
 	return ModelUpdateStr
 }
 
+func (mu *ModelUpdate) MarshalJSON() ([]byte, error) {
+	rtn := make([]map[string]any, 0)
+	for _, u := range *mu {
+		m := make(map[string]any)
+		m[(*u).UpdateType()] = u
+		rtn = append(rtn, m)
+	}
+	return json.Marshal(rtn)
+}
+
+type ModelUpdateItem interface {
+	UpdateType() string
+}
+
 func (update *ModelUpdate) Clean() {
 	if update == nil {
 		return
 	}
-	update.ClientData = update.ClientData.Clean()
+	clientDataUpdates := GetUpdateItems[ClientDataUpdate](update)
+	if len(clientDataUpdates) > 0 {
+		lastUpdate := clientDataUpdates[len(clientDataUpdates)-1]
+		(*ClientData)(lastUpdate).Clean()
+	}
 }
 
-func (update *ModelUpdate) UpdateScreen(newScreen *ScreenType) {
-	if newScreen == nil {
-		return
-	}
-	for idx, screen := range update.Screens {
-		if screen.ScreenId == newScreen.ScreenId {
-			update.Screens[idx] = newScreen
-			return
+func (update *ModelUpdate) append(item *ModelUpdateItem) {
+	*update = append(*update, item)
+}
+
+func AddUpdate[I ModelUpdateItem](update *ModelUpdate, item I) {
```

Comment:
this doesn't need to be generic.  can just do `item ModelUpdateItem`.  could also make this variadic `item ...ModelUpdateItem`.

Reply:
could also add a specific function for setting `Interactive`.  might make some of the code a bit simpler in cmdrunner.